src/server/controllers/users.js

```diff
@@ -149,14 +152,14 @@ module.exports = {
       return User.findOrCreate({where: { email: userObject.email},
         defaults: userObject})
         .spread( async (createdUser, created) => {
+          const userExists = await User.findOne({where: {email: req.body.email}});
           if(!created) {
-            await User.update(req.body,{
-              where: {
-                email: userObject.email
-              },
-              returning: true
-            });
-            return res.status(200).send({ message: 'the user role has been updated' });
+            if(userExists.roleId === req.body.roleId) {
+              const roleName = await matchRoleIdToRoleName(req.body.roleId);
+              return res.status(400).send({ message: `The user with that email address already exists as an ${roleName} . Try updating their role` });
```

Comment:
Seriously, this is a new user we are inviting, we can't use the role in the body. For example if we're inviting the user as an assignee and maybe they are in the DB as admin, they should be notified that they exist as an admin. Your logic will say they exist as assignee.